tutorials/navigation/navigation_using_navigationmeshes.rst

```diff
@@ -370,6 +378,76 @@ The following script uses the NavigationServer to parse source geometry from the
         # Update the region with the updated navigation mesh.
         NavigationServer2D.region_set_navigation_polygon(region_rid, navigation_mesh)
 
+ .. code-tab:: csharp 2D C#
+
+    using Godot;
+
+    public partial class MyNode2D : Node2D
+    {
+        private NavigationPolygon _navigationMesh;
+        private NavigationMeshSourceGeometryData2D _sourceGeometry;
+        private Callable _callbackParsing;
+        private Callable _callbackBaking;
+        private Rid _regionRid;
+
+        public override void _Ready()
+        {
+            _navigationMesh = new NavigationPolygon();
+            _navigationMesh.AgentRadius = 10.0f;
+            _sourceGeometry = new NavigationMeshSourceGeometryData2D();
+            _callbackParsing = Callable.From(OnParsingDone);
+            _callbackBaking = Callable.From(OnBakingDone);
+            _regionRid = NavigationServer2D.RegionCreate();
+
+            // Enable the region and set it to the default navigation map.
+            NavigationServer2D.RegionSetEnabled(_regionRid,true);
+            NavigationServer2D.RegionSetMap(_regionRid,GetWorld2D().GetNavigationMap());
+
+            // Some mega-nodes like TileMap are often not ready on the first frame.
+            // Also the parsing needs to happen on the main-thread.
+            // So do a deferred call to avoid common parsing issues.
+            CallDeferred(MethodName.ParseSourceGeometry);
+        }
+
+        private void ParseSourceGeometry ()
+        {
+            _sourceGeometry.Clear();
+            Node2D rootNode = this;
+
+            // Parse the obstruction outlines from all child nodes of the root node by default.
+            NavigationServer2D.ParseSourceGeometryData(
+                _navigationMesh,
+                _sourceGeometry,
+                rootNode,
+                _callbackParsing
+            );
+        }
+
+        private void OnParsingDone()
+        {
+            // If we did not parse a TileMap with navigation mesh cells we may now only
+            // have obstruction outlines so add at least one traversable outline
+            // so the obstructions outlines have something to "cut" into.
+            _sourceGeometry.AddTraversableOutline(new Vector2[]
+            {
+                new Vector2(0.0f, 0.0f),
+                new Vector2(500.0f, 0.0f),
+                new Vector2(500.0f, 500.0f),
+                new Vector2(0.0f, 500.0f)
```

Comment:
Use comma on all the arrays or none IMO, compare with below

Reply:
Good spot 😁 thanks ill update that, weirdly my ide is stripping the spaces between parameters even though it looks like they are there so ill try update that setting too